core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -130,15 +131,26 @@ private static List<Snapshot> snapshotsWithin(
       if (snapshot.operation().equals(DataOperations.APPEND)) {
         snapshots.add(snapshot);
       } else if (snapshot.operation().equals(DataOperations.OVERWRITE)) {
-        throw new UnsupportedOperationException(
-            String.format(
-                "Found %s operation, cannot support incremental data in snapshots (%s, %s]",
-                DataOperations.OVERWRITE, fromSnapshotId, toSnapshotId));
+        if (incrementalPullIgnoreOverwrite(table)) {
+          continue;
+        } else {
```

Comment:
Nit: the else block isn’t needed. You can remove it since the `continue` statement will move the mlooo forward.

Reply:
Yes, the `else` block is redundant and shall be removed.